flatpak-import.py

```diff
@@ -0,0 +1,212 @@
+
+OSTREE_STATIC_DELTA_SUPERBLOCK_FORMAT = \
+    "(a{sv}tayay" + OSTREE_COMMIT_GVARIANT_STRING + \
+    "aya" + OSTREE_STATIC_DELTA_META_ENTRY_FORMAT + \
+    "a" + OSTREE_STATIC_DELTA_FALLBACK_FORMAT + \
+    ")"
+
+
+if __name__ == "__main__":
+
+    aparser = ArgumentParser(
+        description='Import flatpak into a local repository ',
+    )
+    aparser.add_argument('repo',
+                         help='repository name to use')
+    aparser.add_argument('flatpak', help='file to import')
+    aparser.add_argument('-g', '--gpg-homedir',
+                         help='GPG homedir to use when looking for keyrings')
+    aparser.add_argument('-k', '--keyring', help='additional trusted keyring file')
+    aparser.add_argument('-s', '--sign-key', help='GPG key ID to sign the commit with')
+    aparser.add_argument('-v', '--verbose', action='store_true',
+                         help='verbose log output')
+    aparser.add_argument('-d', '--debug', action='store_true',
+                         help='debug log output')
+
+    config = ConfigParser()
+    config.read('flatpak-import.conf')
+    aparser.set_defaults(**dict(config.items('defaults')))
+
+    args = aparser.parse_args()
+
+    if args.debug:
+        logging.basicConfig(level=logging.DEBUG)
+    elif args.verbose:
+        logging.basicConfig(level=logging.INFO)
+    else:
+        logging.basicConfig(level=logging.WARNING)
+
+
+    ### mmap the flatpak file and create a GLib.Variant from it
+
+    mapped_file = GLib.MappedFile.new(args.flatpak, False)
+    mapped_bytes = mapped_file.get_bytes()
+    ostree_static_delta_superblock_format = GLib.VariantType(
+            OSTREE_STATIC_DELTA_SUPERBLOCK_FORMAT)
+    delta = GLib.Variant.new_from_bytes(
+            ostree_static_delta_superblock_format,
+            mapped_bytes,
+            False)
+
+
+    ### parse flatpak metadata
+
+    # use get_child_value instead of array index to avoid
+    # slowdown (constructing the whole array?)
+    checksum_variant = delta.get_child_value(3)
+    if not OSTree.validate_structureof_csum_v(checksum_variant):
+        # checksum format invalid
+        exit(1)
+
+    metadata_variant = delta.get_child_value(0)
+
+    commit = OSTree.checksum_from_bytes_v(checksum_variant)
+
+    ref = metadata_variant['ref']
+
+    if 'flatpak' in metadata_variant.keys():
+        flatpak = metadata_variant['flatpak']
+    else:
+        flatpak = ''
+
+    if 'origin' in metadata_variant.keys():
+        origin = metadata_variant['origin']
+    else:
+        origin = ''
+
+    if 'runtime-repo' in metadata_variant.keys():
+        runtime_repo = metadata_variant['runtime-repo']
+    else:
+        runtime_repo = ''
+
+    if 'metadata' in metadata_variant.keys():
+        app_metadata = metadata_variant['metadata']
+    else:
+        app_metadata = ''
+
+    if 'gpg-keys' in metadata_variant.keys():
+        gpg_keys = metadata_variant['gpg-keys']
+    else:
+        gpg_keys = ''
```

Comment:
GVariant doesn't make things quite that easy (and conversion to a dictionary is costly in this case), but I moved the individual variables into a dictionary and wrote a small loop to replace this.